src/screens/BackupSheet.js

```diff
@@ -0,0 +1,261 @@
+    selectedWallet,
+    wallets,
+    latestBackup,
+    isWalletLoading,
+  } = useWallets();
+  const [step, setStep] = useState(params?.option || 'first');
+  const walletId = params?.walletId || selectedWallet.id;
+  const missingPassword = params?.missingPassword || null;
+  const { setComponent, hide } = usePortal();
+
+  useEffect(() => {
+    if (isWalletLoading) {
+      setComponent(
+        <LoadingOverlay
+          paddingTop={sheetVerticalOffset}
+          title={isWalletLoading}
+        />,
+        false
+      );
+    }
+    return hide;
+  }, [hide, isWalletLoading, setComponent]);
+
+  const onIcloudBackup = useCallback(async () => {
+    if (latestBackup) {
+      let password = await fetchBackupPassword();
+      // If we can't get the password, we need to prompt it again
+      if (!password) {
+        switchSheetContentTransitionRef.current?.animateNextTransition();
+        setStep(WalletBackupTypes.cloud);
+        setParams({
+          missingPassword: true,
+          option: WalletBackupTypes.cloud,
+        });
+        setOptions({
+          isShortFormEnabled: false,
+          longFormHeight: 10000,
+        });
+        setImmediate(jumpToLong);
+      } else {
+        await dispatch(
+          setIsWalletLoading(walletLoadingStates.BACKING_UP_WALLET)
+        );
+        // We have the password and we need to add it to an existing backup
+        const backupFile = await addWalletToCloudBackup(
+          password,
+          wallets[walletId],
+          latestBackup
+        );
+        if (backupFile) {
+          await dispatch(
+            setWalletBackedUp(walletId, WalletBackupTypes.cloud, backupFile)
+          );
+          logger.log('BackupSheet:: backup saved everywhere!');
+          goBack();
+          setTimeout(() => {
+            Alert.alert('Your wallet has been backed up succesfully!');
+          }, 1000);
+        } else {
+          Alert.alert('Error while trying to backup');
+        }
+      }
+    } else {
+      switchSheetContentTransitionRef.current?.animateNextTransition();
+      setStep(WalletBackupTypes.cloud);
+      setOptions({
+        isShortFormEnabled: false,
+        longFormHeight: 10000,
+      });
+      setImmediate(jumpToLong);
+    }
+  }, [
+    dispatch,
+    goBack,
+    latestBackup,
+    setParams,
+    walletId,
+    wallets,
+    jumpToLong,
+    setOptions,
+  ]);
+
+  const onManualBackup = useCallback(() => {
+    switchSheetContentTransitionRef.current?.animateNextTransition();
+    setStep(WalletBackupTypes.manual);
+    setOptions({
+      isShortFormEnabled: false,
+      longFormHeight: 770,
+    });
+    // wait for layout of sheet
+    setImmediate(jumpToLong);
+  }, [jumpToLong, setOptions]);
+
+  const onIgnoreBackup = useCallback(() => {
+    goBack();
+  }, [goBack]);
+
+  const onAlreadyBackedUp = useCallback(async () => {
+    /// Flag all the wallets as backed up manually
+    Object.keys(wallets).forEach(async walletId => {
```

Comment:
forEach + await <3

Reply:
Actually, maybe we should just expose a function from the wallets redux that does this in one go for all the non-readonly wallets instead of having to await this function for every single wallet and update each, one at a time

Reply:
Regarding this comment, this is what I was thinking: https://github.com/rainbow-me/rainbow/pull/996